udf-compiler/src/main/scala/com/nvidia/spark/udf/State.scala

```diff
@@ -117,23 +118,23 @@ case class State(locals: Array[Expression],
 
 object State {
   def makeStartingState(lambdaReflection: LambdaReflection,
-      children: Seq[Expression]): State = {
+                        children: Seq[Expression],
+                        objref: Option[Expression]): State = {
     val max = lambdaReflection.maxLocals
-    val params: Seq[(CtClass, Expression)] = lambdaReflection.parameters.view.zip(children)
-    val (locals, _) = params.foldLeft((new Array[Expression](max), 0)) { (l, p) =>
-      val (locals: Array[Expression], index: Int) = l
-      val (param: CtClass, argExp: Expression) = p
-
-      val newIndex = if (param == CtClass.doubleType || param == CtClass.longType) {
+    val _children = lambdaReflection.capturedArgs ++ children
```

Comment:
nit, we don't tend to use underscore in variable names. Could call this `capturedAndChildren`?

Reply:
Renamed it to `args`